components/homme/src/theta-l/prim_advance_mod.F90

```diff
@@ -433,21 +463,36 @@ subroutine convert_thermo_forcing(elem,hvcoord,n0,n0q,dt,nets,nete)
         !new q
         qn1(:,:,k) = qn1(:,:,k)/dp(:,:,k)
      enddo
-
+     call get_R_star(rstarn1,qn1)
      tn1(:,:,:) = tn1(:,:,:) + dt*elem(ie)%derived%FT(:,:,:)
 
-     call get_R_star(rstarn1,qn1)
 
-     call get_theta_from_T(hvcoord,rstarn1,tn1,dp,&
-          elem(ie)%state%phinh_i(:,:,:,n0),vthn1)
+     ! update pressure based on Qdp forcing
+     do k=1,nlev
+        ! constant PHI.  FPHI will be zero. cant be used Hydrostatic
+        !pnh(:,:,k) = rstarn1(:,:,k)*tn1(:,:,k)*dp(:,:,k) / &
+        !     ( elem(ie)%state%phinh_i(:,:,k,n0)-elem(ie)%state%phinh_i(:,:,k+1,n0))
+        ! constant NH perturbation pressure
+        pnh(:,:,k)=hvcoord%ps0*hvcoord%hyam(k) + psn1(:,:)*hvcoord%hybm(k) + pprime(:,:,k)
+        exner(:,:,k)=(pnh(:,:,k)/p0)**(Rgas/Cp)
+     enddo
+        
+     ! now we have tn1,dp,pnh - compute corresponding theta and phi:
+     vthn1 =  (rstarn1(:,:,:)/Rgas)*tn1(:,:,:)*dp(:,:,:)/exner(:,:,:)
+
+     phi_n1(:,:,nlevp)=elem(ie)%state%phinh_i(:,:,nlevp,n0)
+     do k=nlev,1,-1
+        phi_n1(:,:,k)=phi_n1(:,:,k+1) + Rgas*vthn1(:,:,k)*exner(:,:,k)/pnh(:,:,k)
+     enddo
 
      !finally, compute difference for FT
      ! this method is using new dp, new exner, new-new r*, new t
      elem(ie)%derived%FT(:,:,:) = &
          (vthn1 - elem(ie)%state%vtheta_dp(:,:,:,n0))/dt
 
+     elem(ie)%derived%FPHI(:,:,:) = &
+         (phi_n1 - elem(ie)%state%phinh_i(:,:,:,n0))/dt
```

Comment:
i am still confused why phi is using forcing and w does not. if one of dyn. eqns, EOS, is used to define forcing on phi, why not use other eqns. for forcing for w?